### DISCUSSION TITLE
Clarify where existing solutions for similar problems fail

### DISCUSSION THREAD

**Discussion by Oskar-van-Deventer** [trustoverip/trust-spanning-protocol#12] [General]
The TSP protocol will be developed to solve problem(s) where current solutions fail. In order to better "sell" the new TSP protocol, we should clarify where existing solutions fail (or are not applicable). Obviously, each and every existing solution, including the "[TSP forerunners](https://github.com/trustoverip/trust-spanning-protocol/discussions/5)" must have at least one problem that they don't solve, neither individually, nor by combining some.

Here is a list of existing solutions, where such clarification may be useful.
-HTTPS
-IPSEC
-TLS
-PGP
-TOR protocol
-NYM and related mixnet technologies
-VPN
-DWN
-KERI
-DIDComm

Oskar

**Comment by talltree** [comment]
@Oskar-van-Deventer This is a brilliant suggestion. I would love—given enough time—to work on a white paper from this task force that provides an analysis of where these protocols/technologies "fail" in achieving the objectives of the TSP. Such a paper would serve as a fabulous introduction to the specification once it is ready.

Now...where to find that time? (Working on it ;-)

I would strongly encourage any TSPTF members who are interested in contributing to such a paper to indicate their interest in a reply to this message.

**Reply by Willem-de-kok** [reply]
I would very much appreciate if the authors of the TSP proposals would spend a bit of time in their presentations (or in upcoming discussions), for as far as that is possible within the time they have, to point out where their proposals solve problems that are not solved by the existing solutions Oskar mentioned. 

I think that could help to assess the strengths and weaknesses of different proposals, both for the proposers as well as for the attenders of the meeting.
@talltree do you think it is reasonable to ask this from the proposers?

**Reply by talltree** [reply]
@Willem-de-kok Yes, I think that is reasonable to ask. It's also something we can focus on as we enter the consolidation phase.   Don't hesitate, Willem, to bring up this question in our upcoming meetings.

Also, I would note that as far as DIDComm and KERI are concerned, this is already happening in depth in the several long discussions threads happening here between Daniel Hardman and Sam Smith. And there was also [one in-depth thread about how TSP will differ from HTTPS/TLS](https://github.com/trustoverip/trust-spanning-protocol/discussions/17#discussioncomment-4908657).
